shared/src/commonMain/kotlin/com/fbada006/shared/composables/ArtMakerDrawScreen.kt

```diff
@@ -162,12 +162,15 @@ internal fun ArtMakerDrawScreen(
 
                         when (event.type) {
                             PointerEventType.Press -> {
-                                getDialogType(change, state.shouldUseStylusOnly)?.let { type ->
+                                // Only update the stylus availability if this is a stylus input and the stylus availability state is not updated
+                                if (change.isStylusInput() && !state.isStylusAvailable) onAction(ArtMakerAction.UpdateStylusAvailability(change.isStylusInput()))
```

Comment:
This is where the detection actually occurs. This should handle both on iOS and android the same now

Reply:
Okay. This makes sense